File: TaxDeductionOrganizer/src/TaxDeductionOrganizer.WebApp/projects/tax-deduction-organizer/src/app/models/deduction.ts
import { DeductionCategory } from './deduction-category';

export interface Deduction {
  deductionId: string;
  taxYearId: string;
  description: string;
  amount: number;
  date: string;
  category: DeductionCategory;
  notes?: string;
  hasReceipt: boolean;
}

export interface CreateDeduction {
  taxYearId: string;
  description: string;
  amount: number;
  date: string;
  category: DeductionCategory;
  notes?: string;
}

export interface UpdateDeduction {
  deductionId: string;
  taxYearId: string;
  description: string;
  amount: number;
  date: string;
  category: DeductionCategory;
  notes?: string;
}

File: TaxDeductionOrganizer/src/Deductions/Deductions.Core/Models/Deduction.cs
namespace Deductions.Core.Models;

public class Deduction
{
    public Guid DeductionId { get; set; }
    public Guid TenantId { get; set; }
    public Guid UserId { get; set; }
    public Guid TaxYearId { get; set; }
    public string Description { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public DateTime Date { get; set; }
    public DeductionCategory Category { get; set; }
    public string? Notes { get; set; }
    public bool HasReceipt { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public void AttachReceipt() => HasReceipt = true;
}

public enum DeductionCategory
{
    Medical,
    Charitable,
    Business,
    Education,
    HomeOffice,
    Travel,
    Other
}
